src/utils.ts

```diff
@@ -61,9 +61,11 @@ export function parseDipState(buf: Buffer) {
   const vaultSplBump = Number(buf.readUInt8(121));
   const optionMint = new PublicKey(buf.slice(122, 154));
   const optionBump = Number(buf.readUInt8(154));
-  const vaultUsdc = new PublicKey(buf.slice(155, 187));
-  const vaultUsdcBump = Number(buf.readUInt8(187));
-  const usdcMint = new PublicKey(buf.slice(188, 220));
+  const vaultPremium = new PublicKey(buf.slice(155, 187));
+  const vaultPremiumBump = Number(buf.readUInt8(187));
+  const premiumMint = new PublicKey(buf.slice(188, 220));
+  const type = String(buf.slice(188, 220));
```

Comment:
This parser handles
https://github.com/Dual-Finance/dual-labs-dip/blob/master/programs/dual-market/src/common.rs